pkg/transforms/http.go

```diff
@@ -38,29 +39,57 @@ type HTTPSender struct {
 	HttpHeaderName      string
 	SecretName          string
 	SecretPath          string
+	URLFormatter        StringValuesFormatter
 }
 
 // NewHTTPSender creates, initializes and returns a new instance of HTTPSender
 func NewHTTPSender(url string, mimeType string, persistOnError bool) HTTPSender {
-	return HTTPSender{
+	return NewHTTPSenderWithOptions(HTTPSenderOptions{
 		URL:            url,
 		MimeType:       mimeType,
 		PersistOnError: persistOnError,
-	}
+	})
 }
 
 // NewHTTPSenderWithSecretHeader creates, initializes and returns a new instance of HTTPSender configured to use a secret header
 func NewHTTPSenderWithSecretHeader(url string, mimeType string, persistOnError bool, headerName string, secretPath string, secretName string) HTTPSender {
-	return HTTPSender{
+	return NewHTTPSenderWithOptions(HTTPSenderOptions{
 		URL:            url,
 		MimeType:       mimeType,
 		PersistOnError: persistOnError,
 		HttpHeaderName: headerName,
 		SecretPath:     secretPath,
 		SecretName:     secretName,
+	})
+}
+
+// NewHTTPSenderWithOptions creates, initializes and returns a new instance of HTTPSender configured with provided options
+func NewHTTPSenderWithOptions(options HTTPSenderOptions) HTTPSender {
+	return HTTPSender{
+		URL:                 options.URL,
+		MimeType:            options.MimeType,
+		PersistOnError:      options.PersistOnError,
+		ContinueOnSendError: options.ContinueOnSendError,
+		ReturnInputData:     options.ReturnInputData,
+		HttpHeaderName:      options.HttpHeaderName,
+		SecretName:          options.SecretName,
+		SecretPath:          options.SecretPath,
+		URLFormatter:        options.URLFormatter,
 	}
 }
 
+type HTTPSenderOptions struct {
+	URL                 string
+	MimeType            string
+	PersistOnError      bool
+	HttpHeaderName      string
+	SecretPath          string
+	SecretName          string
+	URLFormatter        StringValuesFormatter
+	ContinueOnSendError bool
+	ReturnInputData     bool
+}
```

Comment:
Need to add the GoDoc style comments. Be sure to mention that when URLFormatter  is nil, the default from Context is used.